cloudigrade/api/authentication.py

```diff
@@ -175,6 +178,22 @@ def authenticate(self, request):
         """Authenticate the request using the identity header."""
         auth_header, account_number, is_org_admin = parse_requests_header(request)
 
+        if self.support_account_number_header:
+            account_number_header = request.META.get(
+                settings.INSIGHTS_IDENTITY_ACCOUNT_HEADER, None
+            )
+            if account_number_header:
+                account_number = account_number_header
+                logger.info(_("Using account_number '%s' from header"), account_number)
+
+        if self.support_org_admin_header:
+            org_admin_header = request.META.get(
+                settings.INSIGHTS_IDENTITY_ORG_ADMIN_HEADER, None
+            )
+            if org_admin_header:
+                is_org_admin = BooleanField().to_python(org_admin_header)
+                logger.info(_("Using is_org_admin '%s' from header"), is_org_admin)
+
```

Comment:
Per the comment above, I'd suggest removing all of this change. 

Instead, in `parse_requests_header`, where we currently have this:

```python
auth_header = request.META.get(settings.INSIGHTS_IDENTITY_HEADER, None)
```

…I think that could become this:
```python
auth_header = request.META.get(settings.INSIGHTS_IDENTITY_HEADER, None)
if not auth_header and self.allow_internal_fake_identity_header:
    auth_header = request.META.get(settings.INTERNAL_FAKE_INSIGHTS_IDENTITY_HEADER, None)
```

This means we would always try to load the standard header first. If we don't get one _and_ we allow overriding it, then we load the new internal header.

This assumes that we normally get nothing in the standard header for internal requests. I don't know if that's actually the case or if we get something different or partial. My suggestion probably is too simple and needs a little more conditional help.

Maybe we flip the order? If the "allow" flag is true, we could load the new header first, and then if it's still empty, we load the standard header.

```suggestion
```